sdks/python/apache_beam/runners/common.py

```diff
@@ -197,7 +197,11 @@ def receive(self, windowed_value):
   def _dofn_simple_invoker(self, element):
     self._process_outputs(element, self.dofn_process(element.value))
 
-  def _dofn_window_invoker(self, element, args, kwargs, window):
+  def _dofn_per_window_invoker(self, element, args, kwargs):
+    if self.has_windowed_inputs:
```

Comment:
Nice catch. 

I structured it this way to avoid both the looping over windows and construction of a new WindowedValue for the (common) case where there are not multiple windows. It's also more correct by construction as the "window" argument cannot disagree with the "windows" in element (which was the original bug). 

The if statement is repeated, but is very cheap.